src/main/java/sg/dex/starfish/impl/url/ResourceAsset.java

```diff
@@ -27,65 +27,78 @@
+     * This method is to crete Resource Asset with specific resource path, metadata  and isHashOfContentRequired
      *
-     * @param resourceName Resource name
-     * @return ResourceAsset
+     * @param resourcePath            path of the resource
+     * @param isHashOfContentRequired if true then hash of content is calculated and included in metadata.
+     *                                Hash of content will be calculated using Keccak256 hash function
+     *                                if false then content hash is not included in metadata
+     * @return ResourceAsset instance created using given params with default metadata this include DATE_CREATED,TYPE,CONTENT_TYPE
      */
-    public static ResourceAsset create(String resourceName) {
-        return create(buildMetaData(resourceName, null), resourceName);
+    public static ResourceAsset create(String resourcePath, boolean isHashOfContentRequired) {
+        return new ResourceAsset(resourcePath, null, isHashOfContentRequired);
     }
 
     /**
      * This method is to build the metadata of the Resource Asset
      *
      * @param resourcePath resourcePath
-     * @param meta meta
+     * @param meta         meta
      * @return String buildMetadata
      */
-    private static String buildMetaData(String resourcePath, Map<String, Object> meta) {
+    private static String buildMetaData(String resourcePath, String meta, boolean isHashOfContentRequired) {
 
-        InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(resourcePath);
-        if(null ==inputStream){
-            throw new StarfishValidationException("Given Resource path " +resourcePath +" is not valid");
-        }
-        String hash = Hex.toString(Hash.keccak256(Utils.stringFromStream(inputStream)));
 
         Map<String, Object> ob = new HashMap<>();
-        ob.put(Constant.NAME, resourcePath);
         ob.put(Constant.DATE_CREATED, Instant.now().toString());
-        ob.put(Constant.CONTENT_HASH, hash);
         ob.put(Constant.TYPE, Constant.DATA_SET);
         ob.put(Constant.CONTENT_TYPE, "application/octet-stream");
 
+
         if (meta != null) {
-            for (Map.Entry<String, Object> me : meta.entrySet()) {
+
+            for (Map.Entry<String, Object> me : JSON.toMap(meta).entrySet()) {
                 ob.put(me.getKey(), me.getValue());
             }
         }
+
+        if (isHashOfContentRequired) {
+            ob.put(Constant.CONTENT_HASH, getHashOfContent(resourcePath));
+        }
         return JSON.toString(ob);
     }
 
+    private static String getHashOfContent(String resourcePath) {
+        InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(resourcePath);
+        if (null == inputStream) {
+            throw new StarfishValidationException("Given Resource path " + resourcePath + " is not valid");
+        }
+        return Hex.toString(Hash.keccak256(Utils.stringFromStream(inputStream)));
```

Comment:
This doesn't make sense. Why are you translating asset content to a string? It should be considered as a stream of bytes.